core/scheduler/service/replay_worker.go

```diff
@@ -129,6 +154,18 @@ func (w *ReplayWorker) startExecutionLoop(ctx context.Context, replayID uuid.UUI
 
 		// check if replay request is on termination state
 		if syncedRunStatus.IsAllTerminated() {
+			t := replayWithRun.Replay.Tenant()
+			replayState := scheduler.ReplayStateSuccess
+			if syncedRunStatus.IsAnyFailure() {
```

Comment:
lets send the event within `finishReplay` so we don't have to check `IsAnyFailure` twice?